src/server.rs

```diff
@@ -165,162 +165,141 @@ impl<'b> DiagnosticEventHandler<'b> {
         match parsed {
             errors::ParsedError::Diagnostic(d) => {
                 let text = serde_json::to_string(&d).unwrap();
-                self.res.write_all(format!("event: error\ndata: {}\n\n", text).as_bytes())
-                    .unwrap();
-                self.res.flush().unwrap();
+                self.res.set_body(format!("event: error\ndata: {}\n\n", text));
```

Comment:
I don't think it is correct to use `set_body` here - the old write and flush would append the message and send it to the client, whereas `set_body` will replace any old text and not send the data immediately (note that this is an event stream where we stream messages to the client).